subsys/bluetooth/host/audio/ascs.c

```diff
@@ -1011,7 +1357,7 @@ static int ase_metadata(struct bt_ascs_ase *ase, uint8_t op,
 	}
 
 	/* TODO: We should ask the upper layer for accept before we store it */
-	err = bt_audio_ep_set_metadata(ep, buf, meta->len, &ep->codec);
+	err = ascs_ep_set_metadata(ep, buf, meta->len, &ep->codec);
```

Comment:
I'm not sure the upper layers shall accept - I understand metadata as something describing the stream - hence any new metadata received shall overwrite existing metadata (of same type). And then the upper layers shall be notified.
I do see a conflict though if new metadata is defined in the future that is larger that old products expect and overwrites the existing data - that might bring value to the product with new data that the application do not understand.

Reply:
If the client writes metadata that is invalid (e.g. invalid CCID value or something like that), the upper layer shall have the option to reject that.